zappa/handler.py

```diff
@@ -480,6 +480,127 @@ def handler(self, event, context):
+
+                    cookies = []
+                    response_headers = {}
+                    for key, value in response.headers:
+                        if key.lower() == "set-cookie":
+                            cookies.append(value)
+                        else:
+                            if key in response_headers:
+                                updated_value = f"{response_headers[key]},{value}"
+                                response_headers[key] = updated_value
+                            else:
+                                response_headers[key] = value
+
+                    # Calculate the total response time,
+                    # and log it in the Common Log format.
+                    time_end = datetime.datetime.now()
+                    delta = time_end - time_start
+                    response_time_ms = delta.total_seconds() * 1000
+                    response.content = response.data
+                    common_log(environ, response, response_time=response_time_ms)
+
+                    return {
+                        "cookies": cookies,
+                        "isBase64Encoded": response_is_base_64_encoded,
+                        "statusCode": response_status_code,
+                        "headers": response_headers,
+                        "body": response_body,
+                    }
+            except Exception as e:
+                # Print statements are visible in the logs either way
+                print(e)
+                exc_info = sys.exc_info()
+                message = (
+                    "An uncaught exception happened while servicing this request. "
+                    "You can investigate this with the `zappa tail` command."
+                )
+
+                # If we didn't even build an app_module, just raise.
+                if not settings.DJANGO_SETTINGS:
+                    try:
+                        self.app_module
+                    except NameError as ne:
+                        message = "Failed to import module: {}".format(ne.message)
+
+                # Call exception handler for unhandled exceptions
+                exception_handler = self.settings.EXCEPTION_HANDLER
+                self._process_exception(
+                    exception_handler=exception_handler,
+                    event=event,
+                    context=context,
+                    exception=e,
+                )
+
+                # Return this unspecified exception as a 500, using template that API Gateway expects.
+                content = collections.OrderedDict()
+                content["statusCode"] = 500
+                body = {"message": message}
+                if settings.DEBUG:  # only include traceback if debug is on.
+                    body["traceback"] = traceback.format_exception(*exc_info)  # traceback as a list for readability.
+                content["body"] = json.dumps(str(body), sort_keys=True, indent=4)
+                return content
+
```

Comment:
Is this the `else` condition?  If so, it would be clearer to set it as so.

The current code may be a bit difficult to follow.
In some places it returns `content` and in others it returns `zappa_returndict`.

Reply:
@monkut tbh I was just following the existing code for handling the exception. if there's an exception, zappa would return the same structure for v1 and v2 formatted event (dict consist of `body` and `statusCode`)